flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -474,6 +482,19 @@ protected void list(String[] args) throws Exception {
 				System.out.println("--------------------------------------------------------------");
 			}
 		}
+		if (showRemaining) {
+			if (remainingJobs.size() != 0) {
+				remainingJobs.sort(startTimeComparator);
```

Comment:
given the existing pattern of separating by `JobStatus` we may want to do the same for the remaining jobs.

```
runningJobs.stream().collect(Collectors.groupingBy(JobStatusMessage::getJobState))
					.entrySet().stream().sorted(<sort by JobStatus>)
					.map(Map.Entry::getValue).flatMap(List::stream).sorted(<sort by timestamp>).forEachOrdered(<print>);
```

Reply:
Thanks for the reply @zentol , I wasn't sure I fully understand this comment. Are you suggesting that we should always group jobs with same status together during print for all 3 cases?

Reply:
only for the last case, specifically this would separate `FAILED`, `COMPLETED` and `CANCELED` jobs.

That said the other sections aren't that fine-grained either. That reminds me, we could name the last section `terminated` instead of `remaining`.